charts/stable/synapse/templates/_secret.tpl

```diff
@@ -51,7 +51,7 @@ stringData:
           sslmode: "disable"
 
     {{- if .Values.matrix.registration.sharedSecret }}
```

Comment:
```suggestion
    {{- if .Values.matrix.sharedSecret }}
```

Reply:
Moving this back under registration looks correct however that isn't how the shared secret is intended to function. The shared secret is an alternative to the normal registration methods for when you have those disabled.

Reply:
I didn't move it back... You have changed the `registration_shared_secret` value to `.Values.matrix.sharedSecret`, 
but you the if block was still checking the `.Values.registration.sharedSecret`

Reply:
I see, yeah I missed the if